src/support/small_set.h

```diff
@@ -39,16 +40,19 @@ template<typename T, size_t N, typename FlexibleSet> class SmallSetBase {
   size_t usedFixed = 0;
   std::array<T, N> fixed;
 
-  // flexible additional storage
-  FlexibleSet flexible;
+  // flexible additional storage.
+  // Because std::set allocates the root node of its red-black tree on the
+  // heap, we wrap it std::optional to defer that allocation until we start
+  // using the flexible storage.
```

Comment:
This comment is a little specific to the MSVC STL I think. How about this?
```
Flexible additional storage. We use std::optional here to reduce the overhead in the
common case, where the flexible storage is never needed. With std::optional we just
have a pointer taking up space, as opposed to a full std::set which is larger. Also, on
MSVC's STL at least the std::set constructor does work, which we want to avoid here.
```